cli/azd/pkg/executil/builder.go

```diff
@@ -0,0 +1,76 @@
+package executil
+
+import "context"
+
+type Builder struct {
+	commandRunnerFn RunCommandFn
+	runArgs         RunArgs
+}
+
+// Creates a new builder with commandRunnerFn, cmd and args
+func NewBuilder(cmd string, args ...string) *Builder {
```

Comment:
I think the naming here can be improved on. `executil.NewBuilder` is not great for discoverability. `executil.NewRunArgs("a", "b", "c")` seems to be better. Even better if we could rename `executil` -> `exec` and start thinking about it as a proper library instead of util functions

Reply:
Agreed, but still think the `Builder` provider value with all the variations of `RunArgs` that are needed.

Reply:
Merged the builder concept into RunArgs :)